src/vsr/grid_blocks_missing.zig

```diff
@@ -32,30 +32,33 @@ pub const GridBlocksMissing = struct {
 
     const FaultyBlock = struct {
         checksum: u128,
-        progress: FaultProgress,
+        cause: Cause,
         /// Transitions:
         /// - Initial state is `waiting`.
         /// - `waiting → writing` when the block arrives and begins to repair.
         /// - `writing → aborting` when checkpoint becomes durable and the (writing) block is to be
         ///    freed.
         state: enum { waiting, writing, aborting } = .waiting,
-    };
 
-    const FaultProgress = union(enum) {
-        /// Repair a single block.
-        ///
-        /// Originates from one of:
-        /// - the grid scrubber
-        /// - a grid read during prefetch/compaction
-        /// - a grid read while opening the grid/forest
-        block,
-        /// Repair the table and all of its content. Awaiting table index block.
-        table_index: TableIndex,
-        /// Repair the table and all of its content. Awaiting table value blocks.
-        table_value: TableValue,
-
-        const TableIndex = struct { table: *RepairTable };
-        const TableValue = struct { table: *RepairTable, index: u32 };
+        const Cause = union(enum) {
+            /// Repair a single block.
+            ///
+            /// Originates from one of:
+            /// - the grid scrubber
+            /// - a grid read during prefetch/compaction
+            /// - a grid read while opening the grid/forest
+            repair,
+            /// State syncing the index or a value block of a table.
+            /// NB: when a replica decides to sync a block, it might already be repairing.
+            sync: struct {
+                table: *RepairTable,
+                block: union(enum) {
+                    table_index,
+                    /// The index of the value block within the index block.
+                    table_value: u32,
```

Comment:
I named these `table_index`/`table_value` rather than `index`/`value` because `table_value`/`value` has a value which is an index.